backend/typescript/services/implementations/reviewService.ts

```diff
@@ -550,18 +601,75 @@ class ReviewService implements IReviewService {
   async getReviews(
     page: string,
     size: string,
+    genres?: string[],
+    minAge?: number,
+    maxAge?: number,
+    featured?: string,
     searchTerm?: string,
+    draft?: string,
   ): Promise<PaginatedReviewResponseDTO> {
     let result: PaginatedReviewResponseDTO;
 
     try {
       result = await this.db.transaction(async (t) => {
         const limit = size ? parseInt(size, 10) : undefined;
         const offset = this.getPaginationOffset(page, limit);
-        let reviewIds: number[];
 
-        const findConditions: FindAndCountOptions = {
+        // Review Filter Parameters
+        const featureOpt = featured ? { featured } : {};
+        const draftOpt = draft
+          ? {
+              published_at:
+                draft === "true"
+                  ? {
+                      [Op.eq]: null, // We only want drafts
+                    }
+                  : {
+                      [Op.ne]: null, // We only want published reviews
+                    },
+            }
+          : {};
+
+        // Book Filter Parameters
+        let filteredIds: number[] = [];
+        if ((genres && genres.length > 0) || minAge || maxAge) {
+          filteredIds = await this.getReviewIdsFromFilter(
+            genres,
+            minAge,
+            maxAge,
+          );
+        }
+
+        // Search Parameters
+        let searchedIds: number[] = [];
+        if (searchTerm) {
+          searchedIds = await this.getReviewsIdsFromSearch(searchTerm);
+        }
+
+        // If there's search AND filter, we want to get the intersection of the two review_id lists
+        let reviewIds: number[] = [];
+        if (searchedIds.length > 0 && filteredIds.length > 0) {
+          reviewIds = filteredIds.filter((id) => searchedIds.includes(id));
+        } else if (searchedIds.length > 0) {
+          reviewIds = searchedIds;
+        } else if (filteredIds.length > 0) {
+          reviewIds = filteredIds;
+        }
+
+        const searchAndFilterOpt =
```

Comment:
I think we need to handle the case where both `reviewIds` and `searchIds` return nothing. This could happen if we provide a genre that doesn't exist as a filter. Currently, this bypasses the filter altogether and returns all reviews:
![image](https://user-images.githubusercontent.com/19916973/185765412-0b2b8cf4-c24e-4640-97c9-07402b33bc6d.png)
(notice that the returned review doesn't have "random" as its genre).

Maybe one approach to doing this would be to distinguish between null/empty list for `searchIds`, `filterIds`. (ie `null` means param wasn't provided. Empty list means that param was provided and filter/search returned nothing)